src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/connection/ConnectionManager.java

```diff
@@ -252,44 +252,56 @@ private void addShutdownListenerToChannel(Channel channel, Boolean withRecovery)
                         errorBuilder.append(channelCause);
                         String errorString = errorBuilder.toString();
                         LOGGER.warn(errorString);
+
+                        var pinIdToChannelHolder = getChannelHolderByChannel(channel);
+                        if (pinIdToChannelHolder == null) return;
+
                         if (withRecovery && recoverableErrors.stream().anyMatch(errorString::contains)) {
-                            int channelNumber =  channel.getChannelNumber();
-                            var pinIdToChannelHolder = getChannelHolderByChannelNumber(channelNumber);
-                            if ((pinIdToChannelHolder != null) && (pinIdToChannelHolder.getValue().state() != ChannelState.SUBSCRIBING)) {
-                                recoverSubscriptionsOfChannel(channelNumber);
-                            }
+                            var pinId = pinIdToChannelHolder.getKey();
+                            var holder = pinIdToChannelHolder.getValue();
+                            recoverSubscriptionsOfChannel(pinId, channel, holder);
                         }
                     }
                 }
             }
         });
     }
 
-    private @Nullable Map.Entry<PinId, ChannelHolder> getChannelHolderByChannelNumber(int channelNumber) {
+    private @Nullable Map.Entry<PinId, ChannelHolder> getChannelHolderByChannel(Channel channel) {
         return channelsByPin
                 .entrySet()
                 .stream()
-                .filter(entry -> Objects.nonNull(entry.getValue().channel) && channelNumber == entry.getValue().channel.getChannelNumber())
+                .filter(entry -> channel == entry.getValue().channel)
```

Comment:
@lumber1000 could you please advise why did we go back to using the channel here? If this is related to some edge cases or rabbit behavior please leave a comment in the code

Reply:
Because we need holder for exact channel object, not for "channel number N", if channel was changed in holder we don't need to recover it.

Reply:
Got it, thank you. Feel free to leave that as a comment in the code if you think it might help to prevent problems in the future if anybody is going to change the code in this method